### PR TITLE
#gh11573: don't set a BPlocation that isn't marked as shipToDefault/shipTo

### PR DESCRIPTION
**Pull request by adi-stefan**
https://github.com/metasfresh/metasfresh/issues/11573

### CODE REVIEW

File: backend/de.metas.business/src/main/java-legacy/org/compiere/model/MOrder.java [hunk 1]
```diff
@@ -488,10 +488,6 @@ public void setBPartner(final I_C_BPartner bp)
 			}
 		}
 		// set to first
-		if (getC_BPartner_Location_ID() == 0 && !locs.isEmpty())
```
Comment (teosarca) [inline backend/de.metas.business/src/main/java-legacy/org/compiere/model/MOrder.java:491]: pls leave a comment that we are no longer setting to default location if that's not a ship location

File: backend/de.metas.business/src/main/java/de/metas/order/impl/OrderBL.java [hunk 2]
```diff
@@ -684,14 +684,6 @@ public void setBPLocation(final org.compiere.model.I_C_Order order, final org.co
 			{
 				setBPartnerLocation(order, shipLocations.get(0));
 			}
-			else if (!locations.isEmpty())
-			{
-				// set to first
-				if (order.getC_BPartner_Location_ID() == 0)
```
Comment (teosarca) [inline backend/de.metas.business/src/main/java/de/metas/order/impl/OrderBL.java:690]: pls leave a comment that we are no longer setting to default location if that's not a ship location

**Review by teosarca — CHANGES_REQUESTED** [review]
check my comments

File: backend/de.metas.business/src/main/java-legacy/org/compiere/model/MOrder.java (hunk 1 above)
Comment (adi-stefan) [reply backend/de.metas.business/src/main/java-legacy/org/compiere/model/MOrder.java:491]: commented.

File: backend/de.metas.business/src/main/java/de/metas/order/impl/OrderBL.java (hunk 2 above)
Comment (adi-stefan) [reply backend/de.metas.business/src/main/java/de/metas/order/impl/OrderBL.java:690]: commented.

**Pull request merged**